frontend/src/components/bucket/BucketPermissionUser.vue

```diff
@@ -0,0 +1,109 @@
+<script setup lang="ts">
+import { onMounted, ref } from 'vue';
+import { storeToRefs } from 'pinia';
+import Button from 'primevue/button';
+import Dropdown from 'primevue/dropdown';
+import { useBucketStore, useUserStore } from '@/store';
+import type { User } from '@/interfaces';
+
+const userStore = useUserStore();
+
+const { permissions } = storeToRefs(useBucketStore());
+const { userSearch } = storeToRefs(useUserStore());
+
+onMounted(() => {
+  reset();
+});
+
+const isAddDisabled: any = ref(true);
+const showUserExistError: any = ref(false);
+const selectedUser = ref();
+
+const emit = defineEmits(['cancel-bucket-permission-user']);
+
+const cancelBucketPermissionUser = async () => {
+  isAddDisabled.value = true;
+  reset();
+  emit('cancel-bucket-permission-user');
+};
+
+const ifUserExist = () => {
+  return permissions.value.some(e => e.fullName === selectedUser.value.fullName);
+};
+
+const onInput = async (event: any) => {
+  showUserExistError.value = ifUserExist();
+  var inputValue = event.target.value;
```

Comment:
Never use var. Variables must be declared as either a const, or a let if there is intentional reassignment going on.